Gibe.AbTest/AbTestingService.cs

```diff
@@ -15,21 +15,65 @@ public AbTestingService(IAbTestRepository abTestRepository)
 
 		public IEnumerable<Experiment> GetExperiments()
 		{
-			return _abTestRepository.GetExperiments()
+			var experiments = _abTestRepository.GetExperiments()
 				.Select(x => new Experiment(x, GetVariations(x.Id).ToArray()));
+
+			if (experiments.Any())
+			{
+				return experiments;
+			}
+
+			return new[] { EmptyExperiment() };
 		}
 
 		public IEnumerable<Variation> GetVariations(string experimentId)
 		{
-			return _abTestRepository.GetVariations(experimentId)
+			var variations = _abTestRepository.GetVariations(experimentId)
 				.Select(v => new Variation(v));
+
+			if (variations.Any())
+			{
+				return variations;
+			}
+
+			return new[] { EmptyVariation() };
 		}
 
 		public Variation GetVariation(string experimentId, int variationNumber)
 		{
-			var dto = _abTestRepository.GetVariations(experimentId)
-				.First(v => v.VariationNumber == variationNumber);
-			return new Variation(dto);
+			var dtos = _abTestRepository.GetVariations(experimentId);
```

Comment:
First off in the function above we do the same thing variable name variations.

Next this is what FirstOrDefault is for, rather than line 47 to 50. We could do GetVariations(experimentId).FirstOrDefault()